kernel/kernel-api/src/main/java/io/delta/kernel/internal/ActionCommitLog.java

```diff
@@ -0,0 +1,272 @@
+    }
+
+    if (commitVersions.isEmpty()
+        || !Objects.equals(commitVersions.get(commitVersions.size() - 1), expectedEndVersion)) {
+      throw endVersionNotFound(
+          tablePath.toString(),
+          expectedEndVersion,
+          commitVersions.isEmpty()
+              ? Optional.empty()
+              : Optional.of(commitVersions.get(commitVersions.size() - 1)));
+    }
+  }
+
+  /**
+   * Gets an iterator of files in the _delta_log directory starting with the startVersion.
+   *
+   * @throws TableNotFoundException if the directory does not exist
+   */
+  private static CloseableIterator<FileStatus> listLogDir(
+      Engine engine, Path tablePath, long startVersion) {
+    final Path logPath = new Path(tablePath, "_delta_log");
+    try {
+      return wrapEngineExceptionThrowsIO(
+          () ->
+              engine.getFileSystemClient().listFrom(FileNames.listingPrefix(logPath, startVersion)),
+          "Listing from %s",
+          FileNames.listingPrefix(logPath, startVersion));
+    } catch (FileNotFoundException e) {
+      throw new TableNotFoundException(tablePath.toString());
+    } catch (IOException io) {
+      throw new UncheckedIOException("Failed to list the files in delta log", io);
+    }
+  }
+
+  /**
+   * Returns a list of delta commit files found in the _delta_log directory between startVersion and
+   * endVersion (both inclusive).
+   *
+   * @throws TableNotFoundException if the _delta_log directory does not exist
+   */
+  private static List<FileStatus> listCommitFiles(
+      Engine engine, Path tablePath, long startVersion, long endVersion) {
+
+    final List<FileStatus> output = new ArrayList<>();
+    try (CloseableIterator<FileStatus> fsIter = listLogDir(engine, tablePath, startVersion)) {
+      while (fsIter.hasNext()) {
+        FileStatus fs = fsIter.next();
+        if (!FileNames.isCommitFile(getName(fs.getPath()))) {
+          continue;
+        }
+        if (FileNames.getFileVersion(new Path(fs.getPath())) > endVersion) {
+          break;
+        }
+        output.add(fs);
+      }
+    } catch (IOException e) {
+      throw new UncheckedIOException("Unable to close resource", e);
+    }
+    return output;
+  }
+
+  /**
+   * For a table get the list of commit log files for the provided version range.
+   *
+   * @param tablePath path for the given table
+   * @param startVersion start version of the range (inclusive)
+   * @param endVersion end version of the range (inclusive)
+   * @return the list of commit files in increasing order between startVersion and endVersion
+   * @throws TableNotFoundException if the table does not exist or if it is not a delta table
+   * @throws KernelException if a commit file does not exist for any of the versions in the provided
+   *     range
+   * @throws KernelException if provided an invalid version range
+   */
+  public static List<FileStatus> getCommitFilesForVersionRange(
```

Comment:
please put all static methods at the top and private methods below

personally, I prefer to also include

```java
/////////////////
// Public APIs //
/////////////////
```

and the same for private APIs